client/transaction.go

```diff
@@ -4,14 +4,43 @@ import (
 	"bytes"
 	"encoding/json"
 	"github.com/mxenabled/atrium-go/models"
+	"net/http"
 )
 
-func (c *Client) ListTransactions(userGuid string) ([]*models.Transaction, error) {
+func parseTransactionResponse(response *http.Response) (*models.Transaction, error) {
+	if err := parseResponseErrors(response.StatusCode); err != nil {
+		return nil, err
+	}
+
+	buffer := new(bytes.Buffer)
+	buffer.ReadFrom(response.Body)
+	bufferStr := buffer.String()
+	response.Body.Close()
+
+	if response.StatusCode == 200 {
+		transactionResponse := &models.TransactionResponse{}
+		json.Unmarshal([]byte(bufferStr), transactionResponse)
+		return transactionResponse.Transaction, nil
+	}
+
+	return nil, makeGenericError(response.StatusCode, bufferStr)
+}
+
+func (c *Client) ListTransactions(userGuid, fromDate, toDate string) ([]*models.Transaction, error) {
```

Comment:
Let's have a `ListTransactions` and `ListTransactionsWithDateRange` again. And below let's use the param builder.

Btw, since you're doing this param building over and over, you should consider creating a private helper method to build params since private methods are shared within the package namespace.